FireRender.Maya.Src/FireRenderExportCmd.cpp

```diff
@@ -416,171 +418,205 @@ MStatus FireRenderExportCmd::doIt(const MArgList & args)
-		}
+			// process file path
+			std::wstring fileName;
+			std::wstring fileExtension = L"rpr";
+			std::wstring filePath = processedFilePath.asWChar();
 
-		// read file name pattern and padding
-		if (isSequenceExportEnabled && !isAnimationAsSingleFileEnabled && !argData.isFlagSet(kPadding))
-		{
-			MGlobal::displayError("Can't export sequence without setting name pattern and padding!");
-			return MS::kFailure;
-		}
+			// Remove extension from file name, because it would be added later
+			size_t fileExtensionIndex = filePath.find(L"." + fileExtension);
+			bool fileExtensionNotProvided = fileExtensionIndex == -1;
 
-		MString namePattern;
-		argData.getFlagArgument(kPadding, 0, namePattern);
-		unsigned int framePadding = 0;
-		argData.getFlagArgument(kPadding, 1, framePadding);
+			if (fileExtensionNotProvided)
+			{
+				fileName = filePath;
+			}
+			else
+			{
+				fileName = filePath.substr(0, fileExtensionIndex);
+			}
 
-		// create rprs context
-		frw::RPRSContext rprsContext;
+			// append layer name to filename
+			// add support for different layer name suffix formats in the future
+			if (isAllLayersExportEnabled)
+			{
+				MString layerSuffix = "_" + layerNodeFn.name();
+				fileName = fileName + layerSuffix.asWChar();
+			}
 
-		// process each frame
-		for (int frame = firstFrame; frame <= lastFrame; ++frame)
-		{
-			// Move the animation to the next frame.
-			if (isSequenceExportEnabled && !isAnimationAsSingleFileEnabled)
+			// read file name pattern and padding
+			if (isSequenceExportEnabled && !isAnimationAsSingleFileEnabled && !argData.isFlagSet(kPadding))
 			{
-				MTime time;
-				time.setValue(static_cast<double>(frame));
-				MStatus isTimeSet = MGlobal::viewFrame(time);
-				CHECK_MSTATUS(isTimeSet);
+				MGlobal::displayError("Can't export sequence without setting name pattern and padding!");
+				return MS::kFailure;
 			}
 
-			// Refresh the context so it matches the
-			// current animation state and start the render.
-			northStarContextPtr->Freshen();
+			MString namePattern;
+			argData.getFlagArgument(kPadding, 0, namePattern);
+			unsigned int framePadding = 0;
+			argData.getFlagArgument(kPadding, 1, framePadding);
+
+			// create rprs context
+			frw::RPRSContext rprsContext;
 
-			// update file path
-			std::wstring newFilePath; 
-			if (isSequenceExportEnabled && !isAnimationAsSingleFileEnabled)
+			// process each frame
+			for (int frame = firstFrame; frame <= lastFrame; ++frame)
 			{
-				//GetPattern();
-				std::wstring name_regex;
-				std::wstring frame_regex(L"#");
-				std::wstring extension_regex;
-				GetUINameFrameExtPattern(name_regex, extension_regex);
-				std::wstring pattern = namePattern.asWChar();
+				// Move the animation to the next frame.
+				if (isSequenceExportEnabled && !isAnimationAsSingleFileEnabled)
+				{
+					MTime time;
+					time.setValue(static_cast<double>(frame));
+					MStatus isTimeSet = MGlobal::viewFrame(time);
```

Comment:
MGlobal::viewFrame is a heavy command (performance-wise)
I think we should move by frame cycle outside the by layer cycle to avoid extra calls to viewFrame, unless there is a reason not to do so.
Let's discuss